api/repositories/process_repository_test.go

```diff
@@ -4,131 +4,142 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
 )
 
 var _ = Describe("ProcessRepo", func() {
 	var (
-		testCtx      context.Context
-		processRepo  *ProcessRepo
-		namespace    *corev1.Namespace
-		app1GUID     string
-		app2GUID     string
-		process1GUID string
-		process2GUID string
+		testCtx        context.Context
+		processRepo    *ProcessRepo
+		org            *v1alpha2.SubnamespaceAnchor
+		space1, space2 *v1alpha2.SubnamespaceAnchor
+		app1GUID       string
+		app2GUID       string
+		process1GUID   string
+		process2GUID   string
+
+		cfApp1 *workloadsv1alpha1.CFApp
+		cfApp2 *workloadsv1alpha1.CFApp
+
+		cfProcess1 *workloadsv1alpha1.CFProcess
+		cfProcess2 *workloadsv1alpha1.CFProcess
+
+		spaceDeveloperClusterRole *rbacv1.ClusterRole
 	)
 
 	BeforeEach(func() {
 		testCtx = context.Background()
 
-		processRepo = NewProcessRepo(k8sClient)
+		processRepo = NewProcessRepo(k8sClient, userClientFactory)
 
-		namespaceName := generateGUID()
-		namespace = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
-		Expect(
-			k8sClient.Create(testCtx, namespace),
-		).To(Succeed())
+		rootNs := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: rootNamespace}}
+		Expect(k8sClient.Create(testCtx, rootNs)).To(Succeed())
+
+		org = createOrgAnchorAndNamespace(testCtx, rootNamespace, prefixedGUID("org"))
+		space1 = createSpaceAnchorAndNamespace(testCtx, org.Name, prefixedGUID("space1"))
+		space2 = createSpaceAnchorAndNamespace(testCtx, org.Name, prefixedGUID("space2"))
+
+		app1GUID = prefixedGUID("app1")
+		app2GUID = prefixedGUID("app2")
+		process1GUID = prefixedGUID("process1")
+		process2GUID = prefixedGUID("process2")
 
-		app1GUID = generateGUID()
-		app2GUID = generateGUID()
-		process1GUID = generateGUID()
-		process2GUID = generateGUID()
+		spaceDeveloperClusterRole = createClusterRole(testCtx, SpaceDeveloperClusterRoleRules)
 	})
 
 	AfterEach(func() {
 		Expect(
-			k8sClient.Delete(testCtx, namespace),
+			k8sClient.Delete(testCtx, space1),
+		).To(Succeed())
+
+		Expect(
+			k8sClient.Delete(testCtx, space2),
+		).To(Succeed())
+
+		Expect(
+			k8sClient.Delete(testCtx, org),
 		).To(Succeed())
 	})
 
 	Describe("GetProcess", func() {
 		var (
-			namespace1 *corev1.Namespace
-			namespace2 *corev1.Namespace
+			process ProcessRecord
+			getErr  error
 		)
 
 		BeforeEach(func() {
-			namespace1 = namespace
+			cfApp1 = initializeAppCR("test-app1", app1GUID, space1.Name)
+			Expect(k8sClient.Create(context.Background(), cfApp1)).To(Succeed())
+
+			cfApp2 = initializeAppCR("test-app2", app2GUID, space2.Name)
+			Expect(k8sClient.Create(context.Background(), cfApp2)).To(Succeed())
 
-			namespace2Name := generateGUID()
-			namespace2 = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace2Name}}
-			Expect(k8sClient.Create(context.Background(), namespace2)).To(Succeed())
+			cfProcess1 = initializeProcessCR(process1GUID, space1.Name, app1GUID)
```

Comment:
Are there usages of the initializeXXXCR functions that are not followed by `k8sClient.Create`? If not, I would suggest putting CR initialisation and its creation into a single function and internally ensures that cration succeeds - that would spare all the creates in tests.

Reply:
We've created a separate chore to address this across the entire set of repository tests: #647